### DISCUSSION TITLE
Audio Random Access?

### DISCUSSION THREAD

**Discussion by SanJacobs** [free-audio/clap#240] [Q&A]
Does or will clap support ARA (Audio Random Access)?

For those that might not know, ARA is an extension for VST, AU and RTAS that lets the plugin directly read the audio clips that are on in the DAW project.

**Comment by Bremmers** [comment]
The question should be "will ARA support CLAP?". It's up to Celemony to add a CLAP entry point to the ARA SDK.
The other plugin formats don't do anything for ARA, it's ARA that defines a (very simple) extension to the plugin API.

Until recently MultitrackStudio was the only CLAP host that supports ARA. But its ARA implementation is very basic, so there was little reason for Celemony to spend time on this. Now there's also Reaper, so who knows...

It wouldn't be difficult to do by the way.

**Comment by sgretscher** [comment]
Adding ARA to CLAP will be fairly straight forward - it's already being worked on by the community, get in touch with me if you want to participate.

**Comment by defiantnerd** [comment]
The current draft is now in the official ARA repository:

[https://github.com/Celemony/ARA_SDK](https://github.com/Celemony/ARA_SDK)

Feedback is welcome.
